compiler/plc_driver/src/pipelines.rs

```diff
@@ -429,6 +500,10 @@ impl<T: SourceContainer + Sync> AnnotatedProject<T> {
             })
             .collect::<Result<Vec<_>, Diagnostic>>()?;
 
+        if let OnlineChange::Enabled((file, format)) = &compile_options.online_change {
+            write_got_layout(got_layout.into_inner().unwrap(), file, *format)?;
```

Comment:
Unwrapping `std::sync::Mutex::into_inner` does not seem very safe to me - but as you said, since this is hidden behind a feature flag, we can leave it in for now.